src/OffsetArrays.jl

```diff
@@ -370,6 +373,20 @@ end
 
 no_offset_view(A::OffsetArray) = no_offset_view(parent(A))
 
+# Quick hack for matrix multiplication.
+# Ideally, one would instead improve LinearAlgebra's support of custom indexing.
+function Base.:(*)(A::OffsetMatrix, B::OffsetMatrix)
+    matmult_check_axes(A, B)
+    C = OffsetArray(parent(A) * parent(B), (axes(A,1), axes(B,2)))
+end
+
+function Base.:(*)(A::OffsetMatrix, B::OffsetVector)
+    matmult_check_axes(A, B)
+    C = OffsetArray(parent(A) * parent(B), axes(A,1))
+end
+function matmult_check_axes(A, B)
+    axes(A, 2) == axes(B, 1) || error("axes(A,2) = $(UnitRange(axes(A,2))) does not equal axes(B,1) = $(UnitRange(axes(B,1)))")
```

Comment:
Niche picking:

```suggestion
    axes(A, 2) === axes(B, 1) || axes(A, 2) == axes(B, 1) || error("axes(A,2) = $(UnitRange(axes(A,2))) does not equal axes(B,1) = $(UnitRange(axes(B,1)))")
```

gives ~1ns performance improvement.

Reply:
I've implemented this in the latest commit